src/formatDate.js

```diff
@@ -8,7 +8,63 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateObj = parseDate(date, fromFormat);
+
+  return getDate(dateObj, toFormat);
+}
+
+function parseDate(date, fromFormat) {
+  const dateObj = {};
+
+  for (let i = 0; i < 3; i++) {
+    const part = date.split(fromFormat[3])[i];
```

Comment:
Using 'split' directly in a loop can lead to unexpected behavior if the 'fromFormat[3]' changes or is not a string. It's more efficient and safer to split the date once before the loop and work with the resulting array inside the loop.